ws-client/src/lib.rs

```diff
@@ -1,3 +1,9 @@
+#![warn(missing_debug_implementations, missing_docs, rust_2018_idioms, unreachable_pub)]
```

Comment:
I think we should agree on a standard here with @maciejhirsz too. I suggest we make `missing_docs` and `unreachable_pub` errors.

Reply:
I'm happy with `missing_docs` and `unreachable_pub`, `rust_2018_idioms` seems overly pendantic with elided lifetimes (I'd rather have `Cow<str>` than `Cow<'_, str>`, etc.).

Reply:
I think we should keep it as `warn` because `deny/forbid` would force it stop compile on new toolchains as `rustc` gets smarter :)

[forbid(warnings) in substrate](https://github.com/paritytech/substrate/pull/2627/files) is an example of this :)